api/database.go

```diff
@@ -128,6 +130,7 @@ type DatabasePatch struct {
 // DatabaseEdit is the API message for updating a database in UI editor.
 type DatabaseEdit struct {
 	DatabaseID int
+	EngineType db.Type
```

Comment:
We can infer the EngineType from DatabaseID.

Reply:
Done.